generators/server/templates/build.gradle.ejs

```diff
@@ -397,6 +397,11 @@ dependencies {
     <%_ if (['mongodb', 'cassandra', 'couchbase'].includes(databaseType)) { _%>
     implementation "org.springframework.boot:spring-boot-starter-data-<%= databaseType %><% if (reactive) { %>-reactive<% } %>"
     <%_ } _%>
+    <%_ if (databaseType === 'mongodb' && reactive) { _%>
+    // Mongock doesn't support reactive yet, so this dependency is necessary.
+    // See https://github.com/jhipster/generator-jhipster/pull/12536#issuecomment-696683592 for more information.
+    testImplementation "org.springframework.boot:spring-boot-starter-data-mongodb"
```

Comment:
```suggestion
    implementation "org.springframework.boot:spring-boot-starter-data-mongodb"
```

Reply:
I thought Mongock was only for testing. Will fix.